waiter/src/waiter/service_description.clj

```diff
@@ -1571,7 +1571,14 @@
         token-metadata (when token
                          (-> (token->token-parameters kv-store token :error-on-missing false)
                            (attach-token-defaults-fn)
-                           (select-keys token-metadata-keys)))]
+                           (select-keys token-metadata-keys)))
+        unsupported-headers-in-request (set/intersection (set (keys waiter-headers)) unsupported-headers)
+        details {:unsupported-headers-in-request unsupported-headers-in-request}]
```

Comment:
Nitpick: `details` can be moved inside the `when` and more appropriately named to `error-details`.